pennylane/labs/trotter/abstract.py

```diff
@@ -47,3 +56,29 @@ def nested_commutator(fragments: Sequence[Fragment]) -> Fragment:
     head, *tail = fragments
 
     return commutator(head, nested_commutator(tail))
+
+
+class State(ABC):
```

Comment:
Please make sure there is no name duplication. Better to have more specific names than generic names.